test_util/test_azure.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python3
+"""Deploys DC/OS AWS CF template and then runs integration_test.py
+
+The following environment variables control test procedure:
+
+AGENTS: integer (default=2)
+    The number of agents to create in a new cluster.
+
+DCOS_TEMPLATE_URL: string
+    The template to be used for deployment testing
+
+DCOS_NAME: string
+    Instead of providing a template, supply the name (or id) of an already
+    existing cluster
+
+DCOS_SSH_KEY_PATH: string
+    path for the SSH key to be used with a preexiting cluster.
+    Defaults to 'default_ssh_key'
+
+CI_FLAGS: string (default=None)
+    If provided, this string will be passed directly to py.test as in:
+    py.test -vv CI_FLAGS integration_test.py
+
+TEST_ADD_ENV_*: string (default=None)
+    Any number of environment variables can be passed to integration_test.py if
+    prefixed with 'TEST_ADD_ENV_'. The prefix will be removed before passing
+"""
+import logging
+import os
+import sys
+
+from contextlib import ExitStack
+
+from gen.calc import calculate_environment_variable
+from ssh.tunnel import Tunnel
+from test_util.azure import AzureWrapper, TemplateDeployment
+from test_util.helpers import random_id
+from test_util.test_runner import integration_test
+
+LOGGING_FORMAT = '[%(asctime)s|%(name)s|%(levelname)s]: %(message)s'
+logging.basicConfig(format=LOGGING_FORMAT, level=logging.DEBUG)
+log = logging.getLogger(__name__)
+
+
+def check_environment():
+    """Test uses environment variables to play nicely with TeamCity config templates
+    Grab all the environment variables here to avoid setting params all over
+
+    Returns:
+        object: generic object used for cleanly passing options through the test
+
+    Raises:
+        AssertionError: if any environment variables or resources are missing
+            or do not conform
+    """
+    options = type('Options', (object,), {})()
+
+    # Required
+    options.public_ssh_key = calculate_environment_variable('AZURE_PUBLIC_SSH_KEY')
```

Comment:
I wasn't a huge fan of the need to declare the parameter type as a part of the environment name (or at least the way we were doing it). Specifically, oauth enabled is set with a string and vmdiagnostics is set with a bool. Also, I just realized that what I have here will always fail when running against a dcos template instead of a acs tempalte, so what I would like to do is what I already tried for AWS before I realized all params are strings there: pull the given template down, look at the parameters section, verify that all user params are in the template json, cast according the json, and then go ahead.